src/Kdyby/Redis/ConnectionPool.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * This file is part of the Kdyby (http://www.kdyby.org)
+ *
+ * Copyright (c) 2008 Filip Procházka (filip@prochazka.su)
+ *
+ * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
+ */
+
+namespace Kdyby\Redis;
+
+use Kdyby;
+use Nette;
+
+
+
+/**
+ * @author Filip Procházka <filip@prochazka.su>
+ */
+class ConnectionPool extends Nette\Object
+{
+
+	/**
+	 * @var IRedisDriver[]
+	 */
+	private $connections = array();
+
+
+	/**
+	 * Add new connection to pool
+	 * @param string $host
+	 * @param int $port
+	 * @param IRedisDriver $connection
+	 * @throws ConnectionAlreadyInPoolException
+	 */
+	public function addConnection($host, $port, IRedisDriver $connection)
```

Comment:
I think, that this will work also with sockets. You pass host = '/tmp/redis.sock' and port = NULL.